pmda/parallel.py

```diff
@@ -408,43 +405,36 @@ def run(self,
             conclude.elapsed,
             # waiting time = wait_end - wait_start
             np.array([el[4]-wait_start for el in res]),
-            np.array([el[5] for el in res]),
-            np.array([el[6] for el in res]))
+            np.array([el[5] for el in res]))
         return self
 
-    def _dask_helper(self, bslice, indices, top, traj):
+    def _dask_helper(self, bslice, u):
         """helper function to actually setup dask graph"""
         # wait_end needs to be first line for accurate timing
         wait_end = time.time()
-        # record time to generate universe and atom groups
-        with timeit() as b_universe:
-            u = mda.Universe(top, traj)
-            agroups = [u.atoms[idx] for idx in indices]
-
-        res = []
         times_io = []
         times_compute = []
         # NOTE: bslice.stop cannot be None! Always make sure
         #       that it comes from  _trajectory.check_slice_indices()!
-        for i in range(bslice.start, bslice.stop, bslice.step):
+        block_ind = []
+        for i, ts in enumerate(self._universe._trajectory[bslice]):
+            self._frame_index = i
             # record io time per frame
-            with timeit() as b_io:
-                # explicit instead of 'for ts in u.trajectory[bslice]'
-                # so that we can get accurate timing.
-                ts = u.trajectory[i]
-            # record compute time per frame
+            self._ts = ts
             with timeit() as b_compute:
-                res = self._reduce(res, self._single_frame(ts, agroups))
-            times_io.append(b_io.elapsed)
+                self._single_frame()
+
+            block_ind.append(i)
             times_compute.append(b_compute.elapsed)
 
+        #  as oppsed to
+        #  res = []
+        #  for i,ts in traj:
+        #      res.append(self._reduce(...)
+        #  return res
+        #  It does not return the right value except the first block, not totally sure why.
+
         # calculate io and compute time per block
-        return np.asarray(res), np.asarray(times_io), np.asarray(
-            times_compute), b_universe.elapsed, wait_end, np.sum(
+        return np.asarray(self._results[block_ind]), np.asarray(times_io), np.asarray(
+            times_compute), wait_end, np.sum(
             times_io), np.sum(times_compute)
-
-    @staticmethod
-    def _reduce(res, result_single_frame):
-        """ 'append' action for a time series"""
-        res.append(result_single_frame)
-        return res
```

Comment:
How are you going to deal with aggregations as in DensityAnalysis or complicated reductions as in RMSF?

Reply:
It's still doable...with some workaround. i.e. we can do aggregations and further calculations inside `_conclude`, but it also means we have to transfer more data back to the main process. (new examples updated in the gist jupyter notebook)